src/backend/services/wallet-api/src/routes.py

```diff
@@ -31,6 +36,39 @@ def getWalletAmount():
     response = send_get_request( blockchain_wallet_url.format("balance"), data)
     return jsonify(response.json()), response.status_code
 
+@cross_origin()
+@app.route("/history", methods=['GET'])
+def getWalletHistory():
+
+    umnetId = request.args.get('umnetId')
+
+    if umnetId is None:
+        raise IncorrectPayloadException()
+
+    response = requests.get( blockchain_url.format("chain")).json()
+    chain = response['chain']
+
+    result = []
+
+    for block in chain[1:]:
+        block = json.loads(block)
+        ts = float(block["timestamp"])
+        
+        # if user was the miner 
+        if umnetId.upper() == block["miner_id"].upper() :
+            result.append({"Timestamp": ts, "from" : "Mining", "amount": block["reward_amount"]})
+
+        # if user was the sender
+        if umnetId.upper() == block["transaction"]['from_address'].upper():
+            result.append({"Timestamp": ts, "to" : block["transaction"]['to_address'], "amount": block["transaction"]["amount"]})
+        
+        # if user was the reciever
+        if umnetId.upper() == block["transaction"]['to_address'].upper():
+            result.append({"Timestamp": ts, "from" : block["transaction"]['from_address'], "amount": block["transaction"]["amount"]})
```

Comment:
the timestamp for a normal send or receive transaction should be the transaction's timestamp